src/intra_edge.rs

```diff
@@ -16,6 +13,41 @@ pub const EDGE_I420_TOP_HAS_RIGHT: EdgeFlags = 4;
 pub const EDGE_I422_TOP_HAS_RIGHT: EdgeFlags = 2;
 pub const EDGE_I444_TOP_HAS_RIGHT: EdgeFlags = 1;
 
+pub const EDGE_LEFT_HAS_BOTTOM: EdgeFlags =
+    EDGE_I444_LEFT_HAS_BOTTOM | EDGE_I422_LEFT_HAS_BOTTOM | EDGE_I420_LEFT_HAS_BOTTOM;
+pub const EDGE_TOP_HAS_RIGHT: EdgeFlags =
+    EDGE_I444_TOP_HAS_RIGHT | EDGE_I422_TOP_HAS_RIGHT | EDGE_I420_TOP_HAS_RIGHT;
+
+const B: usize = 4;
```

Comment:
which maybe means it can be derived later as  `const B: usize = BlockLevel::COUNT - 1;` , subtracting one because 8x8 is special and not relevant here?

Reply:
I called it `B` like the B in B tree, maybe standing for branching factor or something.  I didn't look closely into why it's 4, though.

Reply:
> ah, I had this confused with the use of `BlockLevel` as an index. If I'm reading the code correctly the 4 is there because when you go from e.g. `16x16` to `32x32`, the number of tiles is multiplied by 4. So even if an extra block level gets added, this constant would just remain the same (it's just a consequence of doing operations in 2D)

Ah, that makes more sense.